contrib/golang/filters/http/source/go/pkg/http/filtermanager.go

```diff
@@ -63,3 +65,254 @@ func getHttpFilterConfigParser(name string) api.StreamFilterConfigParser {
 	}
 	return nil
 }
+
+type filterManagerConfigParser struct {
+}
+
+type filterConfig struct {
+	Name         string
+	parsedConfig interface{}
+}
+
+type filterManagerConfig struct {
+	current []*filterConfig
+}
+
+func (p *filterManagerConfigParser) Parse(any *anypb.Any, callbacks api.ConfigCallbackHandler) (interface{}, error) {
+	conf := &filterManagerConfig{
+		current: []*filterConfig{},
+	}
+	configStruct := &FilterManagerConfig{}
+
+	// No configuration
+	if any.GetTypeUrl() == "" {
+		return conf, nil
+	}
+
+	if err := any.UnmarshalTo(configStruct); err != nil {
+		return nil, err
+	}
+
+	protos := configStruct.GetConfigs()
+	for _, proto := range protos {
+		name := proto.Name
+		if v, ok := httpFilterConfigFactoryAndParser.Load(name); ok {
+			plugin := v.(*filterConfigFactoryAndParser)
+			config, err := plugin.configParser.Parse(proto.Config, nil)
+			if err != nil {
+				return nil, fmt.Errorf("%w during parsing plugin %s in filtermanager", err, name)
+			}
+
+			conf.current = append(conf.current, &filterConfig{
+				Name:         proto.Name,
+				parsedConfig: config,
+			})
+		} else {
+			api.LogErrorf("plugin %s not found, ignored", name)
+		}
+	}
+
+	return conf, nil
+}
+
+func (p *filterManagerConfigParser) Merge(parent interface{}, child interface{}) interface{} {
+	childConfig := child.(*filterManagerConfig)
+
+	// TODO: We have considered to implemented a Merge Policy between the LDS's filter & RDS's per route
+	// config. A thought is to reuse the current Merge method. For example, considering we have
+	// LDS:
+	//	 - name: A
+	//	   pet: cat
+	// RDS:
+	//	 - name: A
+	//	   pet: dog
+	// we will call plugin A's Merge method, which will produce `pet: [cat, dog]` or `pet: dog`.
+	// As there is no real world use case for the Merge feature, I decide to delay its implementation
+	// to avoid premature design.
+	newConfig := *childConfig
+	return &newConfig
+}
+
+type filterManager struct {
+	filters []api.StreamFilter
+
+	callbacks api.FilterCallbackHandler
+}
+
+func filterManagerConfigFactory(c interface{}) api.StreamFilterFactory {
+	conf, ok := c.(*filterManagerConfig)
+	if !ok {
+		panic("unexpected config type")
+	}
+
+	newConfig := conf.current
+	factories := make([]api.StreamFilterFactory, len(newConfig))
+	for i, fc := range newConfig {
+		var factory api.StreamFilterConfigFactory
+		name := fc.Name
+		if v, ok := httpFilterConfigFactoryAndParser.Load(name); ok {
```

Comment:
I'm not sure reusing the same interface here is a good idea.

reasons continue: